clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/Executor.scala

```diff
@@ -42,4 +47,32 @@ object Executor {
       */
     def withFilter(p: A => Boolean): Source[A, M] = source.filter(p)
   }
+
+  /**
+    * Build a stream which, when pulled, will read JSON from a URI and decode it
+    * into the metadata type associated with `index`.
+    */
+  def readMetadata[CI <: ClioIndex](index: ClioIndex)(
```

Comment:
Why in `Executor` instead of in `IoUtil`?

Reply:
There's already a `readMetadata` method in `IoUtil` that just reads file data from a `URI`. Seems poorly named. I'm thinking that this `readMetadata` is pretty specific to the client code and therefore should stay in the `clio-client` codebase.

Reply:
`IoUtil` is in the `clio-client` codebase.

Agreed the existing `readMetadata` is poorly named. It should just be `readFile` or `read`.